StereoKitC/tools/virtual_keyboard.cpp

```diff
@@ -197,13 +191,38 @@ R"(1|2|3|4|5|6|7|8|9|0
 spr:sk/ui/shift---3-go_0|*|=|+|#|%|'|"|spr:sk/ui/backspace-\b--3
 spr:sk/ui/close----close|123---3-go_0|!| - --7|?|Return-\n--4|)";
 	if (virtualkeyboard_parse_layout(layout, strlen(layout), &layer)) keyboard_layers.add(layer);*/
-
-	keyboard_active_root = keyboard_ctx_root_text;
+	keyboard_active_root = text_context_text;
 	keyboard_active_idx  = 0;
 }
 
 ///////////////////////////////////////////
 
+bool virtualkeyboard_switch(text_context_ keyboard_type, char** keyboard_layouts, int layouts_num) {
+	keylayout_info_t layer = {};
+	array_t<keylayout_info_t> keyboard_layers = {};
+	for (int i = 0; i < layouts_num; i++) {
+		if (virtualkeyboard_parse_layout(*(keyboard_layouts + i), (int32_t)strlen(*(keyboard_layouts + i)), &layer)) {
+			keyboard_layers.add(layer);
+		}
+		else {
+			return false;
+		}
+	}
+
+	if (keyboard_layers.count != 0) {
+		switch (keyboard_type) {
+			case text_context_text:  keyboards[0] = keyboard_layers;;   break;
+			case text_context_number: keyboards[1] = keyboard_layers;; break;
+			case text_context_uri:    keyboards[2] = keyboard_layers;;    break;
+			default:                 keyboards[0] = keyboard_layers;;   break;
+		}
```

Comment:
This too, does this still need to be a switch/case, or can it be an `if`?

Reply:
I'd also love to see a bit more attention to detail on spacing and unnecessary syntax. It's fine if it doesn't look exactly like my code, but it should at least be clean and consistent. You've got some ragged whitespace here, as well as unused semicolons.